README.rst

```diff
@@ -40,26 +56,17 @@ openedx-completion-aggregator uses the pluggable django app pattern to ease inst
 
         $ pip install openedx-completion-aggregator
 
-2.  [Optional] You may override the set of registered aggregator block types in your ``lms.yml`` file::
+2.  By default, aggregate data is re-computed synchronously (with each created or updated BlockCompletion). While that is often useful for development, in most production instances, you will want to calculate aggregations asynchronously as explained above. To enable asynchronous calculation for your installation, set the following in your ``lms.yml`` file::
 
         ...
-        COMPLETION_AGGREGATOR_BLOCK_TYPES:
-            - course
-            - chapter
-            - subsection
-            - vertical
+        COMPLETION_AGGREGATOR_ASYNC_AGGREGATION: true
         ...
 
+    Then configure a pair of cron jobs to run ``./manage.py run_aggregator_service`` and ``./manage.py run_aggregator_cleanup`` as often as desired. (Start with hourly and daily, respectively, if you are unsure.) The ``run_aggregator_service`` task is what updates any aggregate completion data values that need to be updated since it was last run. The cleanup task deletes old database entries used to coordinate the aggregation updates, and which can build up over time but are no longer needed.
 
-3.  By default, completion is aggregated synchronously (with each created or updated BlockCompletion). While that is ideal for development, in most production instances, you will want to calculate aggregations asynchronously for better performance.  To enable asynchronous calculation for your installation, set the following in your ``lms.yml`` file::
-
-        ...
-        COMPLETION_AGGREGATOR_ASYNC_AGGREGATION: true
-        ...
+3. A lock is used to prevent overlapping runs of ``run_aggregator_service``. If you have a lot of data on your instance such that a single run of ``run_aggregator_service`` takes longer than 30 minutes, you must configure ``COMPLETION_AGGREGATOR_AGGREGATION_LOCK_TIMEOUT_SECONDS`` to a larger value to prevent releasing the lock too early as a result of a timeout.
 
-    Then configure up a pair of cron jobs to run ``./manage.py
-    run_aggregator_service`` and ``./manage.py run_aggregator_cleanup`` as often
-    as desired.
+4. If the aggregator is installed on an existing instance, then it's sometimes desirable to fill "Aggregate" data for the existing courses. There is the ``reaggregate_course`` management command, which prepares data that will be aggregated during the next ``run_aggregator_service`` run. However, the process of aggregating data for existing courses can place extremely high loads on both your celery workers and your MySQL database, so on large instances this process must be planned with great care. For starters, we recommend you disable any associated cronjobs, scale up your celery worker pool significantly, and scale up your database cluster and storage. To give you some idea, on one Open edX instance this process took 125 hours.
```

Comment:
Side note: it would have taken 125h if we had used 1 worker with its default configuration (1 process). When we ran 12 processes, it took around 11h.

Reply:
OK. I'll just remove that sentence anyways.